src/openms/source/METADATA/SpectrumMetaDataLookup.cpp

```diff
@@ -212,4 +222,254 @@ namespace OpenMS
     return success;
   }
 
+  bool SpectrumMetaDataLookup::addMissingSpectrumReferencestoPeptideIDs(
+    vector<PeptideIdentification>& peptides, const String& filename,
+    bool stop_on_error, bool add_ionmatches, bool reset_basename)
+  {
+    MSExperiment<> exp;
+    SpectrumLookup lookup;
+    bool success = true;
+    String bn = basename(filename.c_str());
+    for (vector<PeptideIdentification>::iterator it = peptides.begin();
+         it != peptides.end(); ++it)
+    {
+      if (reset_basename)
+      {
+        it->setBaseName(bn);
+      }
+      if (!it->metaValueExists("spectrum_reference"))
+      {
+        if (lookup.empty()) // load raw data only if we have to
+        {
+          FileHandler().loadExperiment(filename, exp);
+          lookup.readSpectra(exp.getSpectra());
+        }
+        try
+        {
+          Size index = lookup.findByRT(it->getRT());
+          it->setMetaValue("spectrum_reference", exp[index].getNativeID());
+
+          if (add_ionmatches)
+          {
+            addIonMatches_(*it, exp[index]);
+          }
+        }
+        catch (Exception::ElementNotFound&)
+        {
+          LOG_ERROR << "Error: Failed to look up spectrum reference by retention time for peptide ID with RT'" + String(it->getRT()) + "' - no spectrum with corresponding RT found." << endl;
+          success = false;
+          if (stop_on_error) break;
+        }
+      }
+      else
+      {
+        if (add_ionmatches)
+        {
+          if (lookup.empty()) // load raw data only if we have to
+          {
+            FileHandler().loadExperiment(filename, exp);
+            lookup.readSpectra(exp.getSpectra());
+          }
+          String spectrum_id = it->getMetaValue("spectrum_reference");
+          try
+          {
+            Size index = lookup.findByNativeID(spectrum_id);
+            addIonMatches_(*it, exp[index]);
+          }
+          catch (Exception::ElementNotFound&)
+          {
+            LOG_ERROR << "Error: Failed to look up spectrum reference '" + spectrum_id + "' - for adding IonMatches; no spectrum with corresponding native ID found." << endl;
+            success = false;
+            if (stop_on_error) break;
+          }
+        }
+      }
+    }
+    return success;
+  }
+
+  void SpectrumMetaDataLookup::addIonMatches_(PeptideIdentification& pi, const MSSpectrum<Peak1D>& spec)
```

Comment:
While it is nice to have the possibility of reannotation I would not include it in its current form.
Maybe you can give some details on what you are intending with this function?